packages/gatsby/index.d.ts

```diff
@@ -1119,12 +1204,13 @@ export interface ReplaceComponentRendererArgs extends BrowserPluginArgs {
     path: string
     "*": string
     uri: string
-    location: object
-    navigate: Function
+    location: Location
+    navigate: NavigateFn
     children: undefined
     pageResources: object
     data: object
-    pageContext: object
+    pageContext: { [key: string]: unknown }
```

Comment:
Could these be changed to `Record<string, unknown>`

Reply:
I've made the requested changes